src/MSBuild/TerminalLogger/TerminalLogger.cs

```diff
@@ -944,6 +957,18 @@ private void ThreadProc()
     /// </summary>
     internal void DisplayNodes()
     {
+        if (_buildCanceledEvetnAgrs != null)
+        {
+            if (!_cancellationMessageRendered)
+            {
+                string message = _buildCanceledEvetnAgrs.Message ?? string.Empty;
+                Terminal.WriteLine(message);
+                _cancellationMessageRendered = true;
+            }
+
+            return;
```

Comment:
Why the early return? don't we want to watch the nodes cancel?

Reply:
Offline sync: Use `RenderImmediateMessage`

Reply:
Did you decide not to go to `RenderImmediateMessage`?

Reply:
No, I used `RenderImmediateMessage`